src/lib.rs

```diff
@@ -1,40 +1,209 @@
+//!         //! implement workflow logic here
+//!
+//!         Ok(String::from("success"))
+//!     }
+//!     async fn interact_with_workflow(&self, ctx: SharedWorkflowContext<'_>) -> Result<(), HandlerError> {
+//!         //! implement interaction logic here
+//!         //! e.g. resolve a promise that the workflow is waiting on
+//!
+//!         Ok(())
+//!     }
+//! }
+//!
+//! #[tokio::main]
+//! async fn main() {
+//!     tracing_subscriber::fmt::init();
+//!     HttpServer::new(Endpoint::builder().bind(MyWorkflowImpl.serve()).build())
+//!         .listen_and_serve("0.0.0.0:9080".parse().unwrap())
+//!         .await;
+//! }
+//! ```
+//!
+//! - Specify that you want to create a Workflow by using the [`#[restate_sdk::workflow]` macro](workflow).
+//! - The workflow needs to have a `run` handler.
+//! - The first argument of the `run` handler must be the [`WorkflowContext`](crate::context::WorkflowContext) parameter.
+//!     The `WorkflowContext` parameter is used to interact with Restate.
+//!     The `run` handler executes exactly once per workflow instance.
+//! - The other handlers of the workflow are used to interact with the workflow: either query it, or signal it.
+//!     They use the [`SharedWorkflowContext`](crate::context::SharedWorkflowContext) to interact with the SDK.
+//!     These handlers can run concurrently with the run handler and can still be called after the run handler has finished.
+//! - Have a look at the [workflow docs](workflow) to learn more.
+//!
+//!
+//! Learn more about each service type here:
+//! - [Service](restate_sdk_macros::service)
+//! - [Virtual Object](object)
+//! - [Workflow](workflow)
+//!
+//! # Features
+//!
+//! Have a look at the following SDK capabilities:
+//!
+//! - [Service Communication][crate::context::ContextClient]: Durable RPC and messaging between services (optionally with a delay).
+//! - [Journaling Results][crate::context::ContextSideEffects]: Persist results in Restate's log to avoid re-execution on retries
+//! - State: [read][crate::context::ContextReadState] and [write](crate::context::ContextWriteState): Store and retrieve state in Restate's key-value store
+//! - [Scheduling & Timers][crate::context::ContextTimers]: Let a handler pause for a certain amount of time. Restate durably tracks the timer across failures.
+//! - [Awakeables][crate::context::ContextAwakeables]: Durable Futures to wait for events and the completion of external tasks.
+//! - [Error Handling][crate::errors]: Restate retries failures infinitely. Use `TerminalError` to stop retries.
+//! - [Serialization][crate::serde]: The SDK serializes results to send them to the Server.
+//! - [Serving][crate::http_server]: Start an HTTP server to expose services.
+//!
+//!
+//! ### Logging
+//!
+//! You can set the logging level of the Rust SDK via the `RESTATE_LOGGING` environment variable and the level values can be `TRACE`, `DEBUG`, `INFO`, `WARN` or `ERROR`.
+//! The default log level is `INFO`.
```

Comment:
This is not the full story, we don't embed tracing ourselves.

You need to tell people that the rust SDK uses the `tracing` crate, so you need to configure `tracing_subscriber` to get logs. And then give an example with `tracing_subscriber::fmt::init()` as in one of the examples above, and send them to the tracing subscriber doc: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/index.html#filtering-events-with-environment-variables

Reply:
Can you check if it's okay now?